src/runtime_src/core/common/api/xrt_elf.cpp

```diff
@@ -45,6 +47,16 @@ class elf_impl
       throw std::runtime_error("not a valid ELF stream");
   }
 
+  explicit elf_impl(const void *data, size_t size)
+  {
+    // Uses the same approach as in aiebu reporter.cpp
+    // ibufferstream allows reading from data without first copying over
+    // the data to create the stream
+    boost::interprocess::ibufferstream istr(static_cast<const char *>(data), size);
+    if (!m_elf.load(istr))
+      throw std::runtime_error("not valid ELF data");
+  }
```

Comment:
I think this should work with delegating constructor.
```suggestion
  // Uses the same approach as in aiebu reporter.cpp
  // ibufferstream allows reading from data without first copying over
  // the data to create the stream
  explicit 
  elf_impl(const void *data, size_t size)
    :  elf_impl{boost::interprocess::ibufferstream{static_cast<const_char*>(data), size}}
  {}
```